### DISCUSSION TITLE
Enabling compression for the ADIOS2 backend

### DISCUSSION THREAD

**Discussion by eschnett** [openPMD/openPMD-api#1556] [Q&A] [answered]
I want to use openPMD with ADIOS2 as back-end, and I want to enable ADIOS2 compression via the blosc library. How do I do this?

When I create an ADIOS2 file directly the I would write approximately this:
```C++
      const adios2::Operator compressor =
          adios.DefineOperator("Blosc2Compressor", adios2::ops::LosslessBlosc);
      // Use a high compression rate and a bitshuffle filter
      const adios2::Params compressor_options{
          {adios2::ops::blosc::key::clevel, adios2::ops::blosc::key::clevel_9},
          {adios2::ops::blosc::key::doshuffle,
           adios2::ops::blosc::key::doshuffle_bitshuffle},

...

                    adios2::Variable<CCTK_REAL> var =
                        io.DefineVariable<CCTK_REAL>(varname, {}, {},
                                                     {1, 1, 1});
                    var.AddOperation(compresser, compresser_options);
```

**Comment by franzpoeschel** [comment, marked as answer]
Hey Erik,
these parameters are passed via JSON/TOML, see e.g. [here](https://github.com/openPMD/openPMD-api/blob/9ad33b07a9102abb2cd38ca1c7b524693c16de4e/examples/13_write_dynamic_configuration.cpp#L104) for zlib or [here](https://github.com/openPMD/openPMD-api/blob/9ad33b07a9102abb2cd38ca1c7b524693c16de4e/test/SerialIOTest.cpp#L4891) for Blosc.
You can specify the default compression by passing the parameter to the `Series` constructor or a dataset-specific compression, by setting this JSON value in the `Dataset` class (as passed to `resetDataset()`.

Check also [the documentation](https://openpmd-api.readthedocs.io/en/latest/details/backendconfig.html#adios2).
